app/dojo.py

```diff
@@ -248,3 +251,113 @@ def print_unallocated(self,filename=None):
 			txt_file.write(output)
 			txt_file.close()
 			return ("\nData has been successfully saved to {}.txt\n" .format(filename))
+
+	def reallocate_person(self, emailaddress, new_roomname):
+		"""This method reallocates a person using their unique identifier,
+		in this case their email address, to a the specified new room
+
+		"""
+		new_room = None
+		person_reallocating = None
+		
+		#check if a person with the email address exists
+		for member in itertools.chain(self.all_staff,self.all_fellows):
+			if member.email == emailaddress:
+				person_reallocating = member
+
+		if person_reallocating is None:
+			return (text_format.CRED + "\nCould not find person with email {}!\n"
+				.format(emailaddress)
+				+ text_format.CEND)
+
+		#checks if the room exists
+		for room in itertools.chain(self.all_offices, self.all_livingspace):
+			if room.name == new_roomname:
+				new_room = room	
+
+		if new_room is None:
+			return (text_format.CRED + "\nThe room {} does not exist!\n"
+				.format(new_roomname)
+				+ text_format.CEND)
+
+		#reallocating to a livingspace
+		if new_room.room_type == "LIVING SPACE":
+			if person_reallocating.role == "STAFF":
+				return (text_format.CBOLD +"\nCannot reallocate STAFF to LIVING SPACE!\n" 
+					+text_format.CEND)
+			if len(new_room.occupants) < 4:
+				for this_living_space in self.all_livingspace:
+					for occupant in this_living_space.occupants:
+						if occupant == person_reallocating:
+							current_room = this_living_space
+							if current_room == new_room:
+								return (text_format.CRED + "\nCannot reallocate to the same LIVINGSPACE\n"
+									+ text_format.CEND)
+							else:
+								current_room.occupants.remove(person_reallocating)
+								new_room.occupants.append(person_reallocating)
+								return (text_format.CBOLD + "\n{}-{} was succesfully \
+								 reallocated to {} {}\n".format(person_reallocating.role,
+								  person_reallocating.name,new_room.room_type,new_room.name) 
+								 + text_format.CEND)
+			else:
+				return(text_format.CRED + "\nThe room {} is full! Cannot reallocate {}-{}!\n"
+					.format(new_roomname, person_reallocating.role, person_reallocating.name)
+					+ text_format.CEND)
+
```

Comment:
There is a lot of redundancy between lines 286 to 307 making it difficult to make use of the left half of the page. This means that you will find it hard not to go beyond column 80 as is recommended

`if len(new_room.occupants) == 4`:
      return `blah bla bla`

`if person_reallocating in new_room.occupants:`
`return bla bla`

with those 4 lines, you've gotten rid of about 10 - 13 lines.